benchmark/result/metadata_class.py

```diff
@@ -0,0 +1,41 @@
+"""This module contains the metadata of a benchmark result."""
+
+from datetime import datetime
+
+import git
+from attrs import define, field
+from attrs.validators import instance_of
+
+from baybe import __version__ as baybe_version
+from baybe.serialization.mixin import SerialMixin
+
+
+@define(frozen=True)
+class ResultMetadata(SerialMixin):
+    """The metadata of a benchmark result."""
+
+    benchmark_name: str = field(validator=instance_of(str))
+    """The name of the benchmark."""
+
+    execution_time_sec: float = field(validator=instance_of(float))
+    """The execution time of the benchmark in seconds."""
+
+    gpu_used: bool = field(validator=instance_of(bool))
+    """Whether the benchmark used a GPU."""
+
+    start_datetime: str = field(
+        validator=instance_of(str), factory=datetime.now().isoformat
```

Comment:
Also, why to you already convert to string here? I think it makes more sense to keep an actual datetime object as attribute. Conversion should only happen at the time where it's needed, e.g. for serialzation (but that is then configured in the serializer) or for visualization (which is configured in the plotting utility)